extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/RuntimeBeanProcessor.java

```diff
@@ -82,6 +85,30 @@ public boolean test(BeanInfo bean) {
             producer.returnValue(result);
         }
         classCreator.close();
-        recorder.initSupplierBeans(map);
+        return new BeanInfoBuildItem(staticBeans, runtimeBeans);
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void initStatic(BeanInfoBuildItem infoBuildItem,
+            ArcRecorder recorder) {
+        recorder.initSupplierBeans(infoBuildItem.staticBeans);
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void initRuntime(BeanInfoBuildItem infoBuildItem,
+            ArcRecorder recorder) {
+        recorder.initSupplierBeans(infoBuildItem.runtimeBeans);
```

Comment:
This will overwrite the map for `STATIC_INIT` runtime beans, or?

Reply:
oops